File: 2курс/2семестр/Код/OOP/2/Lab2/Lab2/comorg.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Lab2
{
    internal class ComOrg : Organization
    {
        private string _profit;
        private string _businessType;

        public string Profit
        {
            get { return _profit; }
            set { _profit = value; }
        }
        public string BusinessType
        {
            get { return _businessType; }
            set { _businessType = value; }
        }
        public ComOrg
            (
            int id,
            string name,
            string inn,
            int employees,
            string profit,
            string businessType
            ) : base(id,name,inn,employees)
        {
            _profit = profit;
            _businessType = businessType;
        }
        public ComOrg(string name, string inn, int employees, string profit, string businessType)
           : base(name, inn, employees)
        {
            _profit = profit;
            _businessType = businessType;
        }
        public ComOrg() : base("ООО Дылды", "458358238", 5)
        {
            _profit = "100";
            _businessType = "Розничная торговля";
        }




        public override int Taxes()
        {
            if (string.IsNullOrEmpty(_profit))
                return 0;

            if (int.TryParse(_profit, out int profitValue))
            {
                return profitValue * 20 / 100; // 20 % налог с прибыли
            }
            else
            {
                return 0;
            }
        }

        public override string Report()
        {
            return $"Коммерческая организация: {Name}\n" +
                   $"ИНН: {Inn}\n" +
                   $"Сотрудников: {Employees}\n" +
                   $"Прибыль: {_profit}\n" +
                   $"Тип бизнеса: {_businessType}";
        }
        public string ExpandBusiness()
        {
            int.TryParse(_profit, out int profitValue);
            profitValue = profitValue + profitValue * 20;
            return profitValue.ToString();
        }   
        public string Distribute()
        {
            int.TryParse(_profit, out int profitValue);
            profitValue = profitValue / 22 * 3;
            return profitValue.ToString();
        }
    }
}

File: 2курс/2семестр/Код/ООП/3/laba3/src/main/java/ru/university/lab3/entity/ComOrg.java
package ru.university.lab3.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "comorganizations")
@Data
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
@AllArgsConstructor
public class ComOrg extends Organization {
    private String profit;
    private String businessType;

    public ComOrg(String name, String inn, int employeesCount, String profit, String businessType) {
        this.setName(name);
        this.setInn(inn);
        this.setEmployeesCount(employeesCount);
        this.profit = profit;
        this.businessType = businessType;
    }

    @Override
    public String report() {
        return "Коммерческая организация: " + getName()
                + " | ИНН: " + getInn()
                + " | Тип: " + getBusinessType()
                + " | Сотрудники(по штату): " + getEmployeesCount()
                + " | Прибыль: " + getProfit();
    }

    @Override
    public int taxes() {
        if (this.profit == null || this.profit.isEmpty()) {
            return 0;
        }
        try {
            int value = Integer.parseInt(this.profit);
            return value * 20 / 100; // налог с прибыли
        } catch (NumberFormatException e) {
            return 0; // прибыль не число
        }
    }

    public String expandBusiness() {
        try {
            int profitValue = Integer.parseInt(this.profit);
            profitValue = profitValue + (int) (profitValue * 0.20);
            return String.valueOf(profitValue);
        } catch (NumberFormatException e) {
            return this.profit;
        }
    }

    public String distribute() {
        try {
            int profitValue = Integer.parseInt(this.profit);
            profitValue = profitValue / 22 * 3;
            return String.valueOf(profitValue);
        } catch (NumberFormatException e) {
            return "0";
        }
    }
}